Objects/Converters/ConverterRevit/ConverterRevitShared/Partial Classes/ConvertRebar.cs

```diff
@@ -1,175 +1,332 @@
+          true,
+          true
+        );
       }
-      else if (openCurves.Count > 0)
+      catch (Autodesk.Revit.Exceptions.ArgumentNullException nullEx)
       {
-        //rebar ??= DB.Structure.Rebar.CreateFromCurves(Doc, barType, host, curveLoops, out result);
+        appObj.Update(status: ApplicationObject.State.Failed, logItem: nullEx.Message);
       }
-      else
+      catch (Autodesk.Revit.Exceptions.ArgumentOutOfRangeException rangeEx)
       {
-        appObj.Update(status: ApplicationObject.State.Failed, logItem: "No convertible rebar curves where found.");
-        return appObj;
+        appObj.Update(status: ApplicationObject.State.Failed, logItem: rangeEx.Message);
+      }
+      catch (Autodesk.Revit.Exceptions.ArgumentsInconsistentException inconsistentEx)
+      {
+        appObj.Update(status: ApplicationObject.State.Failed, logItem: inconsistentEx.Message);
       }
+      catch (Autodesk.Revit.Exceptions.ArgumentException argException)
+      {
+        var info =
+          "The element host is not a valid rebar host. -or- The input curves "
+          + "contains at least one curve which is not a bound Line or bound Arc and is not "
+          + "supported for this operation. -or- curves do not form a valid CurveLoop. -or- The "
+          + "input curves contains at least one helical curve and is not supported for this operation.";
+        appObj.Update(status: ApplicationObject.State.Failed, logItem: $"{argException.Message}: {info}");
+      }
+      catch (Autodesk.Revit.Exceptions.DisabledDisciplineException disciplineException)
+      {
+        appObj.Update(status: ApplicationObject.State.Failed, logItem: disciplineException.Message);
+      }
+      finally
+      {
+        if (rebar != null)
+        {
+          SetInstanceParameters(rebar, speckleRebar);
 
-      if (speckleRevitRebar != null)
-        SetInstanceParameters(rebar, speckleRevitRebar);
+          // set additional params
+          if (speckleRebar.barPositions > 0)
+          {
+            rebar.NumberOfBarPositions = speckleRebar.barPositions;
+          }
 
-      appObj.Update(status: ApplicationObject.State.Created, createdId: rebar.UniqueId, convertedItem: rebar);
+          // deleting instead of updating for now!
+          if (docObj != null)
+            Doc.Delete(docObj.Id);
+
+          appObj.Update(status: ApplicationObject.State.Created, createdId: rebar.UniqueId, convertedItem: rebar);
+        }
+      }
       return appObj;
     }
 
-    // need to test to see possible types of curves for rebar
-    private void ProcessRebarCurves(Rebar rebar, out List<DB.Curve> open, out List<DB.CurveArray> closed)
+    private RevitRebarGroup RebarToSpeckle(DB.Structure.Rebar revitRebar)
     {
-      var _open = new List<DB.Curve>();
-      var _closed = new List<DB.CurveArray>();
-
-      foreach (var curve in rebar.curves)
+      // skip freeform rebar for now: not supported by RevitRebarGroup class
```

Comment:
I don't see this skipping would make it into any report?

Reply:
Ah it just returns null which would be marked as a failed conversion in the connector